src/test/java/org/terasology/genome/GenomeNewTest.java

```diff
@@ -0,0 +1,150 @@
+                new Function<String, Integer>() {
+                    @Nullable
+                    @Override
+                    public Integer apply(@Nullable String input) {
+                        return (Character.isUpperCase(input.charAt(0)) ? 1 : 0);
+                    }
+                });
+
+        GenomeDefinition genomeDefinition = new GenomeDefinition(defaultBreedingAlgorithm, genomeMap);
+        GeneIndexGenomeMap genomeMap1 = (GeneIndexGenomeMap) genomeDefinition.getGenomeMap();
+        assertEquals(genomeMap1.propertyDefinitionMap.toString(), genomeMap.propertyDefinitionMap.toString());
+        assertEquals(genomeMap1.getProperty("height", "TT", Integer.class), 1);
+        assertEquals(genomeMap1.getProperty("height", "tt", Integer.class), 0);
+        System.out.println(genomeMap1.getProperty("height", "Tt", Integer.class));
+    }
+
+    @Test
+    void testBreeding() {
+        int counter = 0;
+        String resultGenes = "";
+        int geneIndex = 0;
+        BreedingAlgorithm breedingAlgorithm = defaultBreedingAlgorithm;
+
+        String g1 = "TT";
+        String g2 = "tt";
+        GeneIndexGenomeMap genomeMap = new GeneIndexGenomeMap();
+        int geneIndices[] = new int[]{0, 1};
+        String propertyType = "discrete";
+        genomeMap.addProperty("height", geneIndices, Integer.class, defaultBreedingAlgorithm,
+                new Function<String, Integer>() {
+                    @Nullable
+                    @Override
+                    public Integer apply(@Nullable String input) {
+                        return (Character.isUpperCase(input.charAt(0)) ? 1 : 0);
+                    }
+                });
+
+        GenomeDefinition genomeDefinition = new GenomeDefinition(defaultBreedingAlgorithm, genomeMap);
+        GeneIndexGenomeMap genomeMap1 = (GeneIndexGenomeMap) genomeDefinition.getGenomeMap();
+
+        Map propertyDefinitionMap1 = new LinkedHashMap(genomeMap1.propertyDefinitionMap);
+        ArrayList<GeneIndexGenomeMap.GenePropertyDefinition> genePropertyDefinitions =
+                new ArrayList(propertyDefinitionMap1.values());
+        while (geneIndex != g1.length()) {
+            geneIndices = genePropertyDefinitions.get(counter).geneIndices;
+            breedingAlgorithm = genePropertyDefinitions.get(counter++).breedingAlgorithm;
+            if (geneIndex + geneIndices.length < g1.length()) {
+                resultGenes += "" + breedingAlgorithm.produceCross(g1.substring(geneIndex,
+                        geneIndex + geneIndices.length), g2.substring(geneIndex,
+                        geneIndex + geneIndices.length));
+            } else {
+                System.out.println(resultGenes);
+                resultGenes += "" + breedingAlgorithm.produceCross(g1.substring(geneIndex), g2.substring(geneIndex));
+            }
+            geneIndex += geneIndices.length;
+        }
+        System.out.println(resultGenes);
```

Comment:
This method is not testing anything as there are no assert statements.

Reply:
The resultant genes are randomized according to the breeding algorithms and therefore the print statements were used to check if the result was correct or not. Just the working of this piece of code was enough for the test, but I have replaced it with an assert to check for an empty String at the end